crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -113,17 +113,19 @@ impl OnSettlementEventUpdater {
             .map_err(|err| anyhow!("{}", err))
             .with_context(|| format!("convert nonce {hash:?}"))?;
 
-        let mut auction_id =
-            database::auction_transaction::get_auction_id(&mut ex, &ByteArray(tx_from.0), tx_nonce)
-                .await?
-                .map(AuctionId::Centralized);
+        let mut auction_id = Self::recover_auction_id_from_calldata(&mut ex, &transaction)
+            .await?
+            .map(AuctionId::Colocated);
         if auction_id.is_none() {
             // This settlement either belongs to the other environment (staging, prod) or it
-            // was issued using solver-driver colocation. In that case solvers
-            // are supposed to append the `auction_id` to the calldata.
-            auction_id = Self::recover_auction_id_from_calldata(&mut ex, &transaction)
-                .await?
-                .map(AuctionId::Colocated);
+            // was issued BEFORE solver-driver colocation.
```

Comment:
The comment is now slightly incorrect. Not finding the auction id on the calldata is not related to prod/staging. The only thing we can tell is that at this point is that the auction was issued pre-colocation.
But we could also adjust the current `solver` code to append the auction id as well. I **believe** that should be fine.

Reply:
I initially did this, but reverted the commit, for two reasons:
1. Fix works without appending the `auction_id` to current `solver`.
2. If we do it, then we lose the distinction between `Centralized` and `Colocated` auction ids in the `on_settlement_event_updater.rs` which makes thing really confusing and hard to debug

Reply:
Will adjust the comment

Reply:
Fixed.